course_catalog/etl/youtube.py

```diff
@@ -82,6 +84,67 @@ def get_videos_for_playlist(youtube_client, playlist_id, offered_by):
             break
 
 
+def get_captions_for_video(video):
+    """
+    Function which fetches and returns xml captions for a video object
+
+    Args:
+        video (object) : Video Object
+
+    Returns:
+        a string with the transcript in xml format
+
+    """
+
+    pytube_client = pytube.YouTube(video.url)
+    all_captions = pytube_client.captions.all()
+
+    selected_capiton = next(
```

Comment:
I'd suggest using this as an alternative to using `next()`, `next` is sort-of a lower-level operation, it's something you'll probably rarely use and typically if you find yourself using it there's probably a more Pythonic:

```python
english_captions = [caption for caption in all_captions if caption.code == "en"]
# sort auto-generated captions to the bottom of the list
sorted_captions = sorted(english_captions, key=lambda caption: "auto-generated" in caption.name)

return sorted_captions[0] if sorted_captions else None
```

A few key differences here:

- Only filters out non-english captions once
- The sort uses the property of `bool` values evalating as the integer values 0 and 1 so that means it sorts with auto-generated ones at the bottom
- Since the list is now sorted, we can just grab the first item in the list, if any and return that or `None` if not